cmd/docker/main.go

```diff
@@ -267,7 +269,19 @@ func startServer() *apiserver.Server {
 	mw := middleware.NewVersionMiddleware(version.DockerAPIVersion,
 		version.DockerDefaultVersion,
 		version.DockerMinimumVersion)
+
 	api.UseMiddleware(mw)
+	if vchConfig.HostCertificate.IsNil() {
+		log.Warnf("Docker endpoint running in plain HTTP mode")
+		rdnsNames := vicdns.ReverseLookup(addr)
+		if len(rdnsNames) == 0 {
+			log.Warnf("Could not resolve domain names for %s. Docker endpoint will only be accessible via the IP", addr)
+		}
+
+		rdnsNames[addr] = true // add the IP because that's always allowed
```

Comment:
in debug mode this will be adding:
```
rdnsNames["0.0.0.0"] = true
```
which I think will prevent us from being able to connect at all as the specific IP will not be added and there'll be no rdns lookup for 0.0.0.0.
This will also prevent us hitting the API from the containers (not that I'm certain whether we currently allow that, or should).
I suggest adding `clientIP` explicitly instead of `addr`, and maybe `bridgeIP` depending on current behaviour.

Reply:
OK I'll do some testing for these scenarios